### ISSUE TITLE
Sceduled Advertising

### ISSUE THREAD

**Issue by paul38752** [XtramCZ/auto-advertise-js#14]
Hey, could you add a feature where you can choose times of the day to advertise? I've gotten banned or blacklisted from a few servers because a lot of people are accusing me of being a self-bot. I try my best to blend in as much as possible, but I'm still getting suspected. I try to advertise very infrequently, setting my minimum interval to 30 and my maximum to 400, which would on average post every 3–4 hours, and I also enable double-post spam. This bot has been really useful; I can gain 10–60 members a day, but it looks suspicious to a lot of people. and I can not only get banned, but my servers can be blacklisted there even if I am not a staff member of the servers I am advertising. I would like to set this script to run only during a certain time frame. for example, from 06:00 to 23:00. Maybe even add a feature to randomize the time frame; for example, minimum picks a value between 5 and 8 a.m. and maximum picks a value of 10 p.m. to 12 a.m.

Another feature I would like to suggest is that you only get marked as online while advertising, and after a certain randomized amount of time, you get marked as offline. To avoid suspicion, I keep all my accounts marked as offline, which could also look suspicious.

**Comment by paul38752** [comment]
it actually turns out i set the script to run 1-30 minutes by mistake but again it may still come across as suspicious if i advertise 1am in the morning and 5pm in the day having a feature like this would make me blend in a lot easier

**Comment by XtramCZ** [comment]
Hi, thanks for the suggestion, your idea of minimum and maximum start times is good, but I feel like there are already a lot of "minimum and maximum" config settings, instead, I think only start and end times will work. Since it will still work during the "off" hours but it will **not** send the messages, with the interval set to 3h and allowed start time set to 6 am, it will still send it randomly (e.g. 6-9 am). Do you agree? Otherwise, I would need to check the time every minute and remake the entire script.

**Comment by paul38752** [comment]
Javascript has a gethour function, getHours(), so it should not take too long to implement. I can try to help if you want.
I'm ok with the time being random, but too random looks suspicious. I want it to look as close to a human as possible since people sleep during certain hours. If it runs once on Monday at 1am, 3am, 7am, 12pm, 5pm, and 11pm, then on Tuesday at 5am, 10am, 1pm, 3pm, 7pm, 8pm, and 11pm, it looks suspicious. It's way too inconsistent, and at some point I would have to be sleeping since most humans sleep around the same time, so if I used the different time zone excuse, it would not work.
If I ran the script on Monday at 5 a.m., 7 a.m., and 10 a.m. 3pm, 4pm, 8pm, 10pm, then on Tuesday do 7am, 9am, 12pm, 2pm, 7pm, 9pm. It looks more normal, and I want to blend in as much as possible while being able to advertise constantly.

**Comment by XtramCZ** [comment]
Yeah I understand, but like I said. For example, if you set the "work time" to 7 am-9 pm then it would only send messages from 7-9. My idea is that you could set the start time as 7 am and the script would automatically choose minutes, so you would set only 7 pm and the start time could be 7:30 or 7:41 or 7:05. The end time would be 9 pm so the script could end at 8:40, 8:20, or 9, but that depends on the interval (if the next message would be sent after the end, it will simply not send it and start a "hibernation" process)

What do you think?

**Comment by paul38752** [comment]
yeah thats ok

**Comment by XtramCZ** [comment]
I can't replicate your issue with status, when I run the script I am invisible, don't know why

**Comment by paul38752** [comment]
Xtram chances are you're already makred as offline before you ran that script if you can't make this feature work thats fine its less important most people don't notice it

**Comment by XtramCZ** [comment]
Nope, I was online

**Comment by paul38752** [comment]
Correct me if I'm wrong, but it seems to only work before you run the script, not after, meaning if I set the script to stop at 10, if I ran it at 7 p.m., it would continue after it reached 10 p.m. This could be because it's waiting for the interval and doesn't say until it starts the script again. I'm not sure if you made it so that it would only work after rerunning the script or if it could work during the process.
